docker/compute_worker/compute_worker.py

```diff
@@ -215,22 +215,35 @@ async def watch_file(self):
         last_modified_time = None
         start = time.time()
         expiration_seconds = 60
-        while self.watch:
+
+        last_run = False
```

Comment:
all this `last_run` logic _feels weird_, hmm...

If the point is to allow programs to clean up and send the "final detailed results" can we have a sleep of 5-10 seconds at the end?

Reply:
The point of the `last_run` flag is to make sure the while block gets run at least once after the scoring program finishes. I was running into a case where the scoring program would create the `*.html` file and finish during the 5 second sleep at the end of the while loop, so the `file_path` would never get updated to the v1.5 style. Is there a problem with doing it this way? Maybe we could run a different loop that waits for the file to appear, then it enters this loop?

Reply:
I see, well we could wait 10 seconds after the loop ends to get basically the same behavior?

Reply:
The problem is the path is updated inside the loop block. So the `file_path` doesn't exist, therefore it doesn't enter that `glob.glob` block. Then, the loop will exit, wait 10 seconds, and still not have the correct `file_path`. So we need some way to intelligently wait for one of these files to exist before sending detailed results and exiting the function.